bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+
+            InputStream is = connection.getInputStream();
+            InputStreamReader isReader = new InputStreamReader(is);
+            BufferedReader br = new BufferedReader(isReader);
+            StringBuilder sb = new StringBuilder();
+
+            try {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                String sbString = sb.toString();
+
+                if (StringUtils.startsWith(sbString, "{")) {
+                    return new JSONObject(sbString);
+                } else {
+                    LOG.error("Unable to grab JSON Object. Please ensure URL {} is valid. \nRaw Response: {}", url.toString(), sbString);
+                }
+            } finally {
+                is.close();
+                isReader.close();
+                br.close();
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("JSON successfully fetched for URL '{}'.", url.toString());
+                }
+            }
+        } catch (URISyntaxException e) {
+            LOG.error("URI Syntax Exception {}", e);
+        }
+
+        return new JSONObject();
+    }
+
+    /**
+     * Create or update nodes from remote JSON.
+     * @param json JSONObject
+     * @param parentNode Node
+     * @throws IOException exception
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void createOrUpdateNodes(final JSONObject json, final Node parentNode) throws IOException, JSONException, RepositoryException {
+        Iterator<String> keys = json.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            if (json.get(key) instanceof JSONObject) {
+                if (PROTECTED_NODES.contains(key)) {
+                    continue;
+                }
+
+                String objectPath = String.format("%s/%s", parentNode.getPath(), key);
+                JSONObject objectJson = getJsonFromUri(objectPath);
+                Node node = getOrCreateNode(objectPath, (String) objectJson.get(JcrConstants.JCR_PRIMARYTYPE));
+
+                createOrUpdateNodes(objectJson, node);
+
+                if (DamConstants.NT_DAM_ASSET.equals(parentNode.getProperty(JcrConstants.JCR_PRIMARYTYPE).getValue().getString())) {
+                    node.setProperty("isRemoteAsset", true);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Property 'isRemoteAsset' added for node '{}'.", key, node.getPath());
+                    }
+
+                    // Save and refresh the session after the save refresh count has reached the configured amount.
+                    this.saveRefreshCount++;
+                    if (this.saveRefreshCount == this.remoteAssetsConfig.getSaveInterval()) {
+                        this.session.save();
+                        this.session.refresh(true);
+                        this.saveRefreshCount = 0;
```

Comment:
Id probably move this line (setting to 0) before `.save()` and `.refresh(true)` otherwise if either of those fail the `saveRefreshCount` will forever be greater than the save interval.